lib/client/player.go

```diff
@@ -120,17 +128,26 @@ func (p *sessionPlayer) TogglePause() {
 	}
 }
 
-// RequestStop makes an asynchronous request for the player to stop playing.
-// Playback may not stop before this method returns.
-func (p *sessionPlayer) RequestStop() {
+// EndPlayback makes an asynchronous request for the player to end the playback.
+// Playback might not stop before this method returns.
+func (p *sessionPlayer) EndPlayback() {
 	p.Lock()
 	defer p.Unlock()
 
 	switch p.state {
-	case stateStopped, stateStopping:
-		// do nothing if stop already in progress
+	case stateEnding:
+		// We're already ending, no need to do anything.
+	case stateStopped:
+		// The playRange goroutine has already returned, so we can
+		// signal the end of playback by closing the stopC channel right here.
+		p.close()
 	default:
```

Comment:
nit: Explicitly check and use `default` to error? Seems better to play it safe here.

```suggestion
		case statePlaying:
			// ...		
		default:
			// Log an error, we shouldn't hit this.
```